feature/amazon/src/main/java/com/revenuecat/purchases/amazon/AmazonBackend.kt

```diff
@@ -0,0 +1,84 @@
+package com.revenuecat.purchases.amazon
+
+import com.revenuecat.purchases.PurchasesError
+import com.revenuecat.purchases.common.Backend
+import com.revenuecat.purchases.common.errorLog
+import com.revenuecat.purchases.common.toPurchasesError
+import com.revenuecat.purchases.models.ProductDetails
+import org.json.JSONException
+import org.json.JSONObject
+
+/** @suppress */
+typealias PostAmazonReceiptCallback = Pair<(response: JSONObject) -> Unit, (PurchasesError) -> Unit>
+/** @suppress */
+typealias CallbackCacheKey = List<String>
+
+class AmazonBackend(
+    private val backend: Backend
+) {
+
+    @get:Synchronized @set:Synchronized
+    @Volatile var postAmazonReceiptCallbacks = mutableMapOf<CallbackCacheKey, MutableList<PostAmazonReceiptCallback>>()
+
+    fun getAmazonReceiptData(
+        receiptId: String,
+        appUserID: String,
+        storeUserID: String,
+        productDetails: ProductDetails,
+        onSuccess: (JSONObject) -> Unit,
+        onError: (PurchasesError) -> Unit
+    ) {
+        val cacheKey = listOfNotNull(
+            receiptId,
+            appUserID,
+            storeUserID,
+            productDetails.toString()
+        )
+
+        val body = mapOf(
```

Comment:
this means that it's still a post, right? do we have plans of moving to a get later?

Reply:
I brought it up in the channel the other day, but I will bring it up again today in our weekly meeting.For now, it's still a `POST`